docs/rules/array-callback-return.md

```diff
@@ -70,21 +80,119 @@ var foo = Array.from(nodes, function(node) {
 });
 
 var bar = foo.map(node => node.getAttribute("id"));
+
+myArray.forEach(function(item) {
+    return;
+});
 ```
 
-## Options
+Examples of **correct** code for the `{ "allowImplicit": true }` option:
 
-This rule has an object option:
+```js
+/*eslint array-callback-return: ["error", { allowImplicit: true }]*/
+var undefAllTheThings = myArray.map(function(item) {
+    return;
+});
+```
 
-* `"allowImplicit": false` (default) When set to true, allows implicitly returning `undefined` with a `return` statement containing no expression.
+Examples of **incorrect** code for the `{ "checkForEach": true }` option:
 
-Examples of **correct** code for the `{ "allowImplicit": true }` option:
+```js
+/*eslint array-callback-return: ["error", { checkForEach: true }]*/
+
+myArray.forEach(function(item) {
+    return handleItem(item)
+});
+
+myArray.forEach(function(item) {
+    if (item < 0) {
+        return x;
+    }
+    handleItem(item);
+});
+```
+
+Examples of **correct** code for the `{ "checkForEach": true }` option:
 
 ```js
-/*eslint array-callback-return: ["error", { allowImplicit: true }]*/
+/*eslint array-callback-return: ["error", { checkForEach: true }]*/
+
+myArray.forEach(function(item) {
+    handleItem(item)
+});
+
+myArray.forEach(function(item) {
+    if (item < 0) {
+        return;
+    }
+    handleItem(item);
+});
+
+myArray.forEach(function(item) {
+    handleItem(item);
+    return;
+});
+```
+
+Examples of **incorrect** code for this rule with the `{ "allowImplicit": true, "checkForEach": true }` options:
```

Comment:
I believe I understand now by reading previous conversations why are the docs and tests done this way - at one point it wasn't clear is there some relation between these two options.

I think there is no need for `{ "allowImplicit": true, "checkForEach": true }` (together) examples in the documentation, since we came to conclusion that they're unrelated.

`{ checkForEach: true }` examples already have `return;` so it should be clear that it isn't necessary to set `"allowImplicit": true` in order to allow `return;` in `forEach`.